README.md

```diff
@@ -144,33 +144,112 @@ $ make -C build.assets build-binaries
 
 ### Local Build
 
-To perform a build on your host, ensure you have installed Go. In order to
-include the Rust-powered features like Desktop Access, you'll
-also need `cargo` and `rustc`. The current versions of these tools can be found
-in `build.assets/Makefile`.
+**Important:**
 
-```bash
-# get the source & build:
+* The Go compiler is somewhat sensitive to the amount of memory: you will need
+  **at least** 1GB of virtual memory to compile Teleport. A 512MB instance
+  without swap will **not** work.
+* This will build the latest version of Teleport, **regardless** of whether it
+  is stable. If you want to build the latest stable release, run `git checkout`
+  to the corresponding tag (for example, run `git checkout v8.0.0`) **before**
+  performing a build.
+* The following instructions are Mac-specific.
+
+#### Dependencies
+
+Ensure you have installed correct versions of dependencies:
+
+* `Go` version from
+  [go.mod](https://github.com/gravitational/teleport/blob/master/go.mod#L3)
+  
+  ```shell
+  # if we are not on the latest, you might need to
+  # brew install go@<some_version>, i.e. 1.16
+  #
+  # check which version will be installed by running
+  # brew info go
+  
+  $ brew install go
+  ````
```

Comment:
Homebrew is not heavily used by Go users. Maybe instead link to the Go Download and Install instructions? https://go.dev/doc/install

Reply:
Sure, I can add that. Any particular reason homebrew is not used by Go users? And what is used in favor - just downloading Go and managing versions via `go install ...` ?

Reply:
I have used Go from homebrew for years. 🤷‍♂️

Reply:
Me too, that's usually I what I tell people to do.

If we keep it generic here and move macOS/brew-specific instructions to a separate file, I think this is kind of a moot point. We can just say there "here is a macOS/Homebrew-based setup" and that's it.

Reply:
I moved the instructions into a separate file in https://github.com/gravitational/teleport/pull/15092/commits/66ed09fdf93c03d922bc62cc061752b20a36e098 and provided setup for both homebrew and go.dev download.